Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/function/AzureFunctionMvpModel.java

```diff
@@ -0,0 +1,154 @@
+ */
+
+package com.microsoft.azuretools.core.mvp.model.function;
+
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.appservice.AppServicePlan;
+import com.microsoft.azure.management.appservice.FunctionApp;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.core.mvp.model.AzureMvpModel;
+import com.microsoft.azuretools.core.mvp.model.ResourceEx;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class AzureFunctionMvpModel {
+
+    public static final String CANNOT_GET_WEB_APP_WITH_ID = "Cannot get Function App with ID: ";
+    private final Map<String, List<ResourceEx<FunctionApp>>> subscriptionIdToFunctionApps;
+
+    private AzureFunctionMvpModel() {
+        subscriptionIdToFunctionApps = new ConcurrentHashMap<>();
+    }
+
+    public static AzureFunctionMvpModel getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+
+    public FunctionApp getFunctionById(String sid, String id) throws IOException {
+        Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
+        FunctionApp app = azure.appServices().functionApps().getById(id);
+        if (app == null) {
+            throw new IOException(CANNOT_GET_WEB_APP_WITH_ID + id); // TODO: specify the type of exception.
+        }
+        return app;
+    }
+
+    public void deleteFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().deleteById(appId);
+        subscriptionIdToFunctionApps.remove(sid);
+    }
+
+    public void restartFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).restart();
+    }
+
+    public void startFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).start();
+    }
+
+    public void stopFunction(String sid, String appId) throws IOException {
+        AuthMethodManager.getInstance().getAzureClient(sid).appServices().functionApps().getById(appId).stop();
+    }
+
+    /**
+     * List app service plan by subscription id and resource group name.
+     */
+    public List<AppServicePlan> listAppServicePlanBySubscriptionIdAndResourceGroupName(String sid, String group) {
+        List<AppServicePlan> appServicePlans = new ArrayList<>();
+        try {
+            Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
+            appServicePlans.addAll(azure.appServices().appServicePlans().listByResourceGroup(group));
```

Comment:
are there any reasons of swallow exception?

Reply:
Plan to swallow exception in case blocking UI thread, changed to throw exception in base util method and handle them in caller